FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,227 @@
+                textureCoords.Add(new Vector2((float)j / sides, (float)i / rings));
+                if (i < rings && j < sides)
+                {
+                    uint currentRow = i * (sides + 1);
+                    uint nextRow = (i + 1) * (sides + 1);
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = nextRow + j + 1;
+                    indices[index++] = currentRow + j;
+                    indices[index++] = nextRow + j + 1;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = currentRow + j;
+                }
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "torus", vertices, normals, textureCoords, indices);
+    }
+
+    /// <summary>Returns a shape model after generating it.</summary>
+    public static Model GetModelAfterGenerating(ModelEngine engine, string name, List<Vector3> vertices, List<Vector3> normals, List<Vector2> texCoords, uint[] indices)
+    {
+        Model3DMesh mesh = new()
+        {
+            Name = name,
+            Vertices = [.. vertices.ConvertAll(v => v.ToLocation().ToNumerics())],
+            Normals = [.. normals.ConvertAll(n => n.ToLocation().ToNumerics())],
+            TexCoords = [.. texCoords.ConvertAll(t => new System.Numerics.Vector2(t.X, t.Y))],
+            Indices = indices,
+            Bones = []
+        };
+        Model3D model3D = new()
+        {
+            RootNode = new Model3DNode()
+            {
+                Name = name,
+                Children = [],
+            },            
+            Meshes = [mesh]
+        };
+        return engine.FromScene(model3D, name);
+    }
+
+    /// <summary>Internal methods for shape generation.</summary>
+    public class InternalMethods
+    {
+        /// <summary>Generates a circle and provides the necessary information.</summary>
+        public static void GenerateCircle(List<Vector3> vecs, List<Vector3> norm, List<Vector2> tc, uint[] idx, float radius, uint slices, ref uint currentIdx, uint startIdx, float zC, bool flip)
+        {
+            vecs.Add(new Vector3(0, 0, zC));
+            norm.Add(new Vector3(0, 0, flip ? 1 : -1));
+            tc.Add(new Vector2(0.5f, 0.5f));
+            for (uint i = 0; i <= slices; i++)
+            {
+                float phi = i * 2 * MathHelper.Pi / slices;
+                float sinPhi = (float)Math.Sin(phi);
+                float cosPhi = (float)Math.Cos(phi);
+                float x = radius * cosPhi;
+                float y = radius * sinPhi;
+                float z = zC;
+                vecs.Add(new Vector3(x, y, z));
+                norm.Add(new Vector3(0, 0, flip ? 1 : -1));
+                tc.Add(new Vector2(cosPhi, sinPhi));
+                if (i < slices)
+                {
+                    idx[currentIdx++] = startIdx;
+                    idx[currentIdx++] = startIdx + (flip ? i + 2 : i + 1);
```

Comment:
`startIdx + i + (flip ? 2 : 1);`

Reply:
Reworked in https://github.com/FreneticLLC/FreneticGameEngine/commit/1c7c2494515658292232e3548be7afc2b534064a